File: Exercism Codes/Kotlin/DifferenceOfSquares.kt
class Squares(private val num: Int) {
    fun sumOfSquares(): Int {
        return (1..num).sumOf { it * it }
    }
    fun squareOfSum(): Int {
        return (1..num).sum() * (1..num).sum()
    }
    fun difference(): Int {
        return squareOfSum() - sumOfSquares()
    }
}

File: Exercism Codes/JavaScript/difference-of-squares.js
export class Squares {
    constructor(n) {
        this.n = n;
    }
    get sumOfSquares() {
        return [...new Array(this.n + 1).keys()]
            .reduce((a, c) => a + (c ** 2), 0);
    }
    get squareOfSum() {
        return [...new Array(this.n + 1).keys()]
            .reduce((a, c) => a + c, 0) ** 2;
    }
    get difference() {
        return this.squareOfSum - this.sumOfSquares;
    }
}
